client/src/components/game/ruler/Ruler.vue

```diff
@@ -173,7 +213,17 @@ export default {
     },
     onRulerPointCreated (e) {
       this.points.push(e)
-
+      if (e.type == 'carrier' && this.points.length == 1) {
+        this.speedModifier = 1;
+        if (e.object.specialistId ) {
+          for (let i = 0; i < this.carrierSpecialists.length - 1; i++) {
+            if ( this.carrierSpecialists[i].id == e.object.specialistId  && this.carrierSpecialists[i].modifiers.local.speed ) {
+              this.speedModifier = this.carrierSpecialists[i].modifiers.local.speed
```

Comment:
I may be wrong but I'm sure carriers already have the specialist object on them.

So `e.object.specialist.something` may already be accessible.

If so, the speed modifier could simply be assigned to the specialist speed value if there is one.